src/geopackage/read.cpp

```diff
@@ -118,8 +135,12 @@ std::shared_ptr<geojson::FeatureCollection> geopackage::read(
         max_y = bbox[3] > max_y ? bbox[3] : max_y;
     }
 
-    return std::make_shared<geojson::FeatureCollection>(
+    auto fc = std::make_shared<geojson::FeatureCollection>(
         std::move(features),
         std::vector<double>({min_x, min_y, max_x, max_y})
     );
+
+    fc->update_ids();
```

Comment:
This is the only line here I'm curious about, since it's a deeper change in behavior.

That function doesn't have any sort of docstring, and I can't immediately intuit what the cross-reference data structure it's inserting into is for. I see this was added in a commit describing it as a bug fix, but there's not enough detail for me to understand it. Maybe an occasion to document it?

Reply:
Oooh... I will have to take a look and/or ask @program-- to comment... he gave me that part of the PR in a patch. I believe it's something just missed from the geojson analogue though.

Reply:
I can't say exactly the purpose, but it seems like the `feature_by_id` member was used as a way to try and keep O(1) access to features and also keep them in a vector? I'm not sure _why_, but the reason this was needed in this PR was that something changed that caused a different `geojson::FeatureCollection` constructor be called, and not all of the constructors place the given features into the `feature_by_id` map.

Reply:
Ok, I'll take that as given for now. Maybe we'll rethink these data structures down the line, or at least be able to stop having to fiddle with how they're implemented so much.